src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassReportServiceImpl.java

```diff
@@ -0,0 +1,302 @@
+        writeProposalDetailsToReport(rassBatchJobReport);
+        reportWriterService.writeNewLines(1);
+        writeAwardsDetailsToReport(rassBatchJobReport);
+        reportWriterService.writeNewLines(1);
+        reportWriterService.writeSubTitle("*** End of Report Details ***");
+        reportWriterService.writeNewLines(1);
+    }
+
+    private void writeAgencyDetailsToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Agency> agencyResultGrouping = processingResults.getAgencyResults();
+        List<RassBusinessObjectUpdateResult<Agency>> agencyResults = agencyResultGrouping.getObjectResults();
+        writeResultsDetailsToReport(agencyResults, Agency.class, true);
+    }
+
+    private void writeProposalDetailsToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Proposal> proposalResultGrouping = processingResults.getProposalResults();
+        List<RassBusinessObjectUpdateResult<Proposal>> proposalResults = proposalResultGrouping.getObjectResults();
+        writeResultsDetailsToReport(proposalResults, Proposal.class, false);
+    }
+
+    private void writeAwardsDetailsToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Award> awardResultGrouping = processingResults.getAwardResults();
+        List<RassBusinessObjectUpdateResult<Award>> awardResults = awardResultGrouping.getObjectResults();
+        writeResultsDetailsToReport(awardResults, Award.class, true);
+    }
+
+    private <R extends PersistableBusinessObject> void writeResultsDetailsToReport(List<RassBusinessObjectUpdateResult<R>> results, Class businessObjectClass,
+            boolean shouldReportUpdates) {
+        writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.SUCCESS_NEW);
+        if (shouldReportUpdates) {
+            writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.SUCCESS_EDIT);
+        }
+        writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.ERROR);
+        writeResultsDetailsToReportByResultCode(results, businessObjectClass, RassConstants.RassObjectUpdateResultCode.SKIPPED);
+    }
+
+    private <R extends PersistableBusinessObject> void writeResultsDetailsToReportByResultCode(List<RassBusinessObjectUpdateResult<R>> results,
+            Class businessObjectClass, RassObjectUpdateResultCode resultCode) {
+        List<RassBusinessObjectUpdateResult<R>> filteresdResults = results.stream().filter(result -> resultCode.equals(result.getResultCode()))
```

Comment:
Minor typo in variable name; should be changed from "filteresdResults" to "filteredResults" (this is a second instance of the same typo)

Reply:
corrected